### DISCUSSION TITLE
How to build a big quadcopter with Crazyflie Bolt 1.0

### DISCUSSION THREAD

**Discussion by mfxjx000205** [bitcraze/discussions#959] [Q&A] [answered]
Hi everyone,

We want to build a big quadcopter with crazyflie bolt 1.0. However, as a team in the field of computer science, we cannot exactly unserstand the principle of build a big quadcopter. We have read the blog: [https://www.bitcraze.io/2020/10/testing-crazyflie-bolt-and-1-cell-li-ion/](url) but still feel puzzle.

By reading this blog, we believe the hardest thing is how to find a correct ESC and make them work well. So, I want to ask questions below:

1. How should I choose a correct ESC? 

2. After installing ESC, how should I modify the code in crazyflie firmware? I find motor.c, but I'm not sure if this needs to be modified or what other.

Best wishes~

**Comment by tobbeanton** [comment, marked as answer]
> How should I choose a correct ESC?

Do you need any special kind of ESC? Else I would use the one we used in the blog post.

> After installing ESC, how should I modify the code in crazyflie firmware? I find motor.c, but I'm not sure if this needs to be modified or what other.

The default official Bolt build should work with most ESCs (OneShot125 is default protocol). If you want to build on your own do
`make bolt_defconfig` to get the Bolt platform and its defaults. Then you can do `make menuconfig`if you would like to alter any build settings such as ESC protocol.

**Reply by mfxjx000205** [reply]
@tobbeanton 

Hi！

Sorry for late reply! After listening to your excellent advice, I've followed the blog as written and try to install the crazyflie bolt this week. I have already done most of steps. However, I still have a small question: how to charge the Li-ion battery?

 I try to connect the SONY 18650 battery to XT30U-F.But after I soldered it, the original charger no longer worked. **How should I charge the battery**? Is it using a USB cable connected to crazyflie to charge it?(The datasheet said that Don’t have the battery and the USB connected at the same time.
)

Best Wishes!

**Reply by knmcguire** [reply]
hi! since you selected an answer already, you should start a new discussion, or unselect the answer to avoid confusion.

**Reply by mfxjx000205** [reply]
OK，I have unselected the answer，Thank you！

**Reply by tobbeanton** [reply]
You could create a adapter cable and use our "PCB" charger. Since it charges with 500mA it will take long time tough. Perhaps invest in a LiPo/LiIon charger with a XT30 connector?
